test/e2e/replica_set_tls_upgrade/replica_set_tls_upgrade_test.go

```diff
@@ -53,7 +53,8 @@ func TestReplicaSetTLSUpgrade(t *testing.T) {
 	t.Run("MongoDB is reachable while TLS is being enabled", func(t *testing.T) {
 		defer tester.StartBackgroundConnectivityTest(t, time.Second*10, WithoutTls())()
 		t.Run("Upgrade to TLS", tlstests.EnableTLS(&mdb, true))
-		t.Run("Stateful Set Reaches Ready State, after enabling TLS", mongodbtests.StatefulSetIsReady(&mdb))
+		t.Run("Stateful Set Leaves Ready State, after setting TLS to preferSSL", mongodbtests.StatefulSetBecomesUnready(&mdb))
+		t.Run("Stateful Set Reaches Ready State, after setting TLS to preferSSL", mongodbtests.StatefulSetIsReady(&mdb))
```

Comment:
Can we standardize the names here? Maybe replace `StatefulSetIsReady` with `StatefulSetBecomesReady`?